src/stores/reservation.js

```diff
@@ -0,0 +1,58 @@
+import { defineStore } from "pinia";
+import { retrieveSeance } from "@/api/useSeancesApi.js";
+import { retrieveHall } from "@/api/useHallsApi.js";
+import { retrieveMovie } from "@/api/useMoviesApi.js";
+import { formatMovieLength } from "@helpers/useFormatMovieLength";
+
+export const useReservationStore = defineStore({
+  id: "reservation",
+  state: () => ({
+    seance: null,
+    hall: null,
+    movie: null,
+  }),
+  getters: {
+    hallMatrix() {
+      if (this.seance) {
+        const seats = [
+          ...this.seance.taken_seats.map((seat) => ({
+            seat,
+            taken: true,
+            row: seat.slice(0, 1),
+            column: +seat.slice(1),
+          })),
+          ...this.seance.available_seats.map((seat) => ({
+            seat,
+            taken: false,
+            row: seat.slice(0, 1),
+            column: +seat.slice(1),
+          })),
+        ];
+        const hallRows = seats.reduce((rows, seat) => {
+          const _rows = { ...rows };
```

Comment:
I don't think it's necessary to make a copy of this 🤔

Reply:
Agreed